File: typescript/testing/src/jobQueue.ts
export class JobQueue {
  #jobs: string[] = [];

  enqueue(job: string): number {
    this.#jobs.push(job);
    return this.#jobs.length;
  }

  next(): string | undefined {
    return this.#jobs.shift();
  }

  clear(): void {
    this.#jobs = [];
  }

  get size(): number {
    return this.#jobs.length;
  }
}

File: python/testing/src/job_queue.py
from __future__ import annotations


class JobQueue:
    def __init__(self) -> None:
        self._jobs: list[str] = []

    def enqueue(self, job: str) -> int:
        self._jobs.append(job)
        return len(self._jobs)

    def next(self) -> str | None:
        if not self._jobs:
            return None

        return self._jobs.pop(0)

    def clear(self) -> None:
        self._jobs = []

    @property
    def size(self) -> int:
        return len(self._jobs)
